### PR TITLE
Escape JSON keys

### PR DESCRIPTION
**Pull request by RubenRBS**
JSON keys need to also be escaped.

Fixes compiler-explorer/compiler-explorer/issues/3963

### CODE REVIEW

File: src/utils/jsonwriter.cpp [hunk 1]
```diff
@@ -64,6 +38,32 @@ std::string escape(const std::string_view in)
     return out;
 }
 
+void AsmParser::JsonWriter::writeKeyName(const char *key)
+{
+    this->out << "\"" << escape(key) << "\": ";
```
Comment (RubenRBS) [inline src/utils/jsonwriter.cpp:43]: The change is in this line and #48. The rest comes from moving the `escape(std::string_view)` function to the top of the file before the constructor

**Comment by partouf** [comment]
I wonder if we should actually escape the `char*` interface, I don't know how optimized the escaping will be while the work doesn't actually needs to be done.

**Comment by partouf** [comment]
> I wonder if we should actually escape the `char*` interface, I don't know how optimized the escaping will be while the work doesn't actually needs to be done.

Seems not at all, the std::string is always created https://godbolt.org/z/6Yn6nxhMv

**Comment by partouf** [comment]
can't make it constexpr because of the allocation, but we can make it `static`, and then it's mostly a no-op https://godbolt.org/z/zxs9YbY9s

**Comment by partouf** [comment]
OH oops, maybe not, it does something else in when using the actual workflow/cmake buildsettings https://godbolt.org/z/5qafKP4qb

**Comment by partouf** [comment]
Let's just remove the escaping for `const char *`, would have to resort to template shenanigans to make a compile-time checkable thing.

File: src/utils/jsonwriter.cpp (hunk 1 above)
Comment (RubenRBS) [inline src/utils/jsonwriter.cpp:43]:
```suggestion
    this->out << "\"" << key << "\": ";
```

**Pull request merged**